### PR TITLE
Added GetCustomAttributes method to ModuleInfo (#20345)

### LINKED ISSUE #20345: Add metdata API to get module custom attributes
### Rationale

System.Reflection.Metadata offers convenience methods to access custom attributes on most of the record types that correspond to `HasCustomAttribute` token types in the ECMA-335 spec. Notable omission is the `ModuleDefinition` record.

### Proposed API

```csharp
namespace System.Reflection.Metadata
{
    public struct ModuleDefinition
    {
        public CustomAttributeHandleCollection GetCustomAttributes();
    }
}
```

### Other thoughts

This can be worked around with `MetadataReader.GetCustomAttributes(System.Reflection.Metadata.Ecma335.MetadataTokens.EntityHandle(0x1))`, but it's rather awkward.

### PR DESCRIPTION
**Pull request by cod7alex**
Adds `ModuleInfo.GetCustomAttributes` method according to approved API in #20345. Closes #20345.

@tmat i guess you will review :)

### CODE REVIEW

File: src/System.Reflection.Metadata/tests/Metadata/MetadataReaderTests.cs [hunk 1]
```diff
@@ -2122,12 +2114,24 @@ public void ValidateCustomAttribute()
...
+                new byte[] { 0x01, 00, 0x10, 0x4d, 0x6f, 0x64, 0x56, 0x42, 0x44, 0x65, 0x66, 0x61, 0x75, 0x6c, 0x74, 0x50, 0x72, 0x6f, 0x70, 00, 00 }
+            };
+            
+            var reader = GetMetadataReader(NetModule.ModuleVB01, true);
+            var moduleDefinition = reader.GetModuleDefinition();
 
-            i = 0;
-            foreach (var caHandle in reader.CustomAttributes)
+            int i = 0;
+            foreach (var caHandle in moduleDefinition.GetCustomAttributes())
```
Comment (ViktorHofer) [inline src/System.Reflection.Metadata/tests/Metadata/MetadataReaderTests.cs]: nit: I tend to use a for loop if I need a counter variable.

File: src/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeSystem/ModuleDefinition.cs [hunk 2]
```diff
@@ -55,5 +55,10 @@ public GuidHandle BaseGenerationId
                 return _reader.ModuleTable.GetEncBaseId();
             }
         }
+
+        public CustomAttributeHandleCollection GetCustomAttributes()
+        {
+            return _reader.CustomAttributes;
```
Comment (nguerrera) [inline src/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeSystem/ModuleDefinition.cs]: This is incorrect. It is going to return all custom attributes in the custom attribute table, not just those parented by the module definition. The code here should mirror the equivalent method on AssemblyDefinition, with EntityHandle.ModuleDefinition replacing EntityHandle.AssemblyDefinition.

File: src/System.Reflection.Metadata/tests/Metadata/MetadataReaderTests.cs (hunk 1 above)
Comment (nguerrera) [inline src/System.Reflection.Metadata/tests/Metadata/MetadataReaderTests.cs]: Where possible, I tend to agree, but in this case we have a collection that isn't indexable IIRC.

File: src/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeSystem/ModuleDefinition.cs (hunk 2 above)
Comment (cod7alex) [inline src/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeSystem/ModuleDefinition.cs]: thanks!

**Comment by cod7alex** [comment]
I have changed the method, but none of test resources have module attributes to test.

**Comment by danmosemsft** [comment]
@nguerrera could you have another look

**Pull request merged**